### PR TITLE
add JS test for alias register and revoke

### PR DESCRIPTION
**Pull request by cryptix**
This is based on #85 which should be merged first.

I wanted to add a test for aliases to it but couldn't.

1) The client plugin needs a new version after https://github.com/ssb-ngi-pointer/ssb-room-client/pull/1
2) even with master of the above I get this:

```
/home/cryptix/ssb/ssb-room-client/lib/plugin-room-client.js:30
                rooms.get(roomKey).rpc.room.registerAlias(alias, sig, cb);

TypeError: Cannot read property 'registerAlias' of undefined
    at Object.registerAlias (/home/cryptix/ssb/ssb-room-client/lib/plugin-room-client.js:30:44)
    at Object.hooked [as registerAlias] (/home/cryptix/go-repos/ssb-rooms/muxrpc/test/nodejs/node_modules/hoox/index.js:10:15)
    at Timeout._onTimeout (/home/cryptix/go-repos/ssb-rooms/muxrpc/test/nodejs/testscripts/modern_aliases.js:12:23)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
```

I suspect some namespace/manifest meh-ness. I successfully used `room.registerAlias` already with `go-ssb/cmd/sbotcli` which ignores the manifest.

### CODE REVIEW

**Comment by cryptix** [comment]
@staltz would be cool if you could take this.

I think this just needs an updated package.json in `muxrpc/test/nodejs`, updating the deps and running `go test -run Aliases` should then pass, I think.

**Comment by staltz** [comment]
@cryptix I updated ssb-room-client (I think in the future I'd like to update secret-stack to support a nicer way of declaring remote APIs that are not implemented locally) and other dependencies, but now I think there are errors from Go:

```
revokeAlias: not your alias (moderators need to use the web dashboard of the room
```

Also, registerAlias should respond with a string matching the alias URL. This should also be asserted in tape.

File: muxrpc/test/nodejs/testscripts/modern_aliases.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  secretStackPlugins,
+  before,
+  after: (t, sbot, rpc, exit) => {
+
+    // give ssb-conn a moment to settle
+    setTimeout(() => {
+
+      sbot.roomClient.registerAlias(rpc.id, "alice", (err, ret) => {
+        t.error(err, 'registerAlias')
+        t.comment(`registerAlias value: ${ret}`)
```
Comment (staltz) [inline muxrpc/test/nodejs/testscripts/modern_aliases.js:14]:
```suggestion
        t.ok(ret)
        t.equals(typeof ret, 'string')
        t.ok(new URL(ret))
```

**Comment by cryptix** [comment]
Thank you! I can take this over again.

> I think there are errors from Go

There is a missing mock-setup call to have it not flunk out here:

https://github.com/ssb-ngi-pointer/go-ssb-room/blob/07feacb5587e2baa912b8203909cf41fa6f0a672/muxrpc/handlers/alias/handler.go#L118

(background: these tests run with a mocked alias store so it doesn't actually store the alias)

**Comment by cryptix** [comment]
All done and register now returns the URL.

Fixes #86

**Review by staltz — APPROVED** [review]
Looks great

**Pull request merged**